### PR TITLE
A new non-autoload Music API

### PR DESCRIPTION
**Pull request by simonknowsstuff**
- New music api with automatic playlist features that lets you load up a few songs and let the api do the rest
- You can now toggle your HUD by F1

### CODE REVIEW

File: HUD/HUD.gd [hunk 1]
```diff
@@ -1,6 +1,12 @@
 extends CanvasLayer
 
-signal stream_ready(stream)
+var MusicAPI
+
+func _ready(): MusicAPI = get_tree().get_nodes_in_group("MusicAPI")[0]
```
Comment (Technohacker) [inline HUD/HUD.gd:5]: This feels kinda convoluted, I'll have this changed to an autoload :)

**Review by Technohacker — CHANGES_REQUESTED** [review]
Overall, looks good :)

File: HUD/HUD.gd (hunk 1 above)
Comment (simonknowsstuff) [reply HUD/HUD.gd:5]: Noted, making changes asap!

Comment (simonknowsstuff) [reply HUD/HUD.gd:5]: Fixed autoload!

**Review by Technohacker — APPROVED** [review]

**Pull request merged**